src/main/java/org/isf/medicalstockward/manager/MovWardBrowserManager.java

```diff
@@ -295,4 +295,31 @@ public int compare(MovementForPrint o1, MovementForPrint o2) {
 		}
 	}
 
+	/* * 
+	 * Delete the MovementWard for the specified criteria.
+	 *
+	 * @param movWard - the movement ward
+	 * @throws OHServiceException if an error occurs retrieving the medical.
+	 */
+	public void deleteLastMovementWard(MovementWard movWard) throws OHServiceException {
```

Comment:
I think we should enforce the check if the specified movement is really the last movement at the manager level, so also API can leverage this check.

(same also for OP-1020, I forgot to mention this)

Reply:
Did you check any possible circumstance where the last ward movement deletion can be denied?

Reply:
yes but until now I don't found. but I have add the check if the sppecified movement is really the last movement

Reply:
ok, thank you. Let me test it thoroughly. Did you test with the dump method, even involving the main pharmacy?

Reply:
Indeed, I think there is a problem upon discharging to another ward and then cancel the movement: the medical is added to the destination ward and then not removed

Reply:
Differently from Main Store where we have only positive quantities but different movement type, in the ward pharmacies we have a positive movements for the discharge to patient/internal/ward and negative movements for incoming from other wards; in addition, `oh_medicaldsrward` keeps track of in and out quantities for each medical and lot, for quick stock on hand calculation.

Reply:
okay tky